### PR TITLE
add OCR Decoding support - WIP

### PR DESCRIPTION
**Pull request by N950**
Adding Nodes/Loss/Metric for OCR recognition/decoding
Conformity to contribution guidelines will be fixed before closing

### CODE REVIEW

File: luxonis_train/attached_modules/losses/ocr_decoding_loss.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import torch
+from torch import Tensor, nn
+
+from .base_loss import BaseLoss
+
+
+class FocalCTC(BaseLoss[Tensor, Tensor]):
```
Comment (kozlov721) [inline luxonis_train/attached_modules/losses/ocr_decoding_loss.py:7]: Add class attribute `supported_labels = [LabelType.TEXT`]

File: luxonis_train/nodes/ocr_decoder.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import torch
+import torch.nn as nn
+import torchvision
+from torch import Tensor
+
+from .base_node import BaseNode
+from luxonis_train.utils.types import LabelType
+
+
+class ResBlock(nn.Module):
```
Comment (kozlov721) [inline luxonis_train/nodes/ocr_decoder.py:17]: I believe both `ResBlock` nad `UpSample` should already be implemented in `luxonis_train.nodes.blocks` or in one of the backbones

File: luxonis_train/nodes/ocr_decoder.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        self.block = nn.Sequential(
+            nn.Conv2d(in_channels=ch_in, out_channels=ch_out, kernel_size=kernel_size, stride=stride, padding=padding)
+        )
+
+    def forward(self, x):
+        out = self.block(x)
+        return out
+
+
+class OCRDecoderBackbone(BaseNode):
```
Comment (kozlov721) [inline luxonis_train/nodes/ocr_decoder.py:51]: Instead of `_task_type` argument, add a class attribute `tasks: list[LabelType] = [LabelType.TEXT]`

File: luxonis_train/attached_modules/losses/ocr_decoding_loss.py (hunk 1 above)
Comment (kozlov721) [reply luxonis_train/attached_modules/losses/ocr_decoding_loss.py:7]: Also the metrics and losses will probably require custom `prepare` method that takes the packed output of the head and labels and prepares argument to be passed to `forward` or `update`.
